src/main/java/org/activiti/cloud/qa/story/QueryCampainActions.java

```diff
@@ -34,11 +37,36 @@
     @Then("the tweet should be in the list of the processed tweets")
     public void theMatchingTweetIsInTheList() {
         await().untilAsserted(() -> {
-            List<Tweet> matchingTweets = queryCampaignSteps.getProcessedTweets();
-            assertThat(matchingTweets).isNotNull();
+            Collection<Resource<Tweet>> matchingTweetsRaw = queryCampaignSteps.getProcessedTweets().getContent();
+            assertThat(matchingTweetsRaw).isNotNull();
+
+            List<Tweet> matchingTweets = new ArrayList<>();
+            for( Resource<Tweet> resource: matchingTweetsRaw){
```

Comment:
No need to create this temporary list here. You can change the expression of `extracting` method directly:
```java
assertThat(matchingTweetsRaw)
   .extracting(tweetResource -> tweetResource.getContent().getText())
   .contains(TweeterConnectorActions.getLastSentMatchingTweet().getText());
```

Reply:
Understood 👍